pkg/utils/integration/snapshots.go

```diff
@@ -0,0 +1,136 @@
+package integration
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/devfile/library/v2/pkg/util"
+	appstudioApi "github.com/redhat-appstudio/application-api/api/v1alpha1"
+	"github.com/redhat-appstudio/e2e-tests/pkg/utils"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// CreateSnapshotWithImage creates a snapshot using an image.
+func (i *IntegrationController) CreateSnapshotWithImage(applicationName, namespace, componentName, containerImage string) (*appstudioApi.Snapshot, error) {
+	hasSnapshot := &appstudioApi.Snapshot{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "snapshot-sample-" + util.GenerateRandomString(4),
+			Namespace: namespace,
+			Labels: map[string]string{
+				"test.appstudio.openshift.io/type":           "component",
+				"appstudio.openshift.io/component":           componentName,
+				"pac.test.appstudio.openshift.io/event-type": "push",
+			},
+		},
+		Spec: appstudioApi.SnapshotSpec{
+			Application: applicationName,
+			Components: []appstudioApi.SnapshotComponent{
+				{
+					Name:           componentName,
+					ContainerImage: containerImage,
+				},
+			},
+		},
+	}
+	err := i.KubeRest().Create(context.TODO(), hasSnapshot)
+	if err != nil {
+		return nil, err
+	}
+	return hasSnapshot, err
+}
+
+// CreateSnapshotWithCompoents creates a Snapshot using the given parameters.
+func (i *IntegrationController) CreateSnapshotWithComponents(name string, namespace string, applicationName string, snapshotComponents []appstudioApi.SnapshotComponent) (*appstudioApi.Snapshot, error) {
```

Comment:
This function is similar with CreateSnapshotWithImage() above. Can the name of the parameters and the order be adjust to be more unified?

Reply:
Sure! Changed to match order.

Reply:
@jinqi7 In my opinion, the name of the resource (if we're explicitly supplying it) that is getting created by the function should be the first in the list of parameters. I propose to revert this change unless you have some strong opinions against it. Wdyt?

Reply:
With looking into the two functions, I think the function CreateSnapshotWithImage() can call CreateSnapshotWithComponents() to avoid the confusion and duplication. Wdyt?

Reply:
BTW,  "snapshotName" is better than "name" in the parameter.

Reply:
Changed order of parameters to snapshotName first and removed code duplication in CreateSnapshot...() functions.